src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramBuilder.cpp

```diff
@@ -1938,13 +1958,24 @@ class ShaderWriteDepth : public ShaderPart
 						;
 				} else {
 					if (_glinfo.isGLESX && config.generalEmulation.enableClipping != 0) {
-						m_part =
-							"highp float writeDepth()						        										\n"
-							"{																								\n"
-							"  if (uDepthSource != 0) return uPrimDepth;													\n"
-							"  return clamp(8.0 * (gl_FragCoord.z * 2.0 - 1.0) * uDepthScale.s + uDepthScale.t, 0.0, 1.0);	\n"
-							"}																								\n"
-						;
+						if (_glinfo.isGLESX && _glinfo.noPerspective) {
```

Comment:
Just 
`if (_glinfo.noPerspective)`
`_glinfo.isGLESX` is checked above

Reply:
Tep, thanks.